project/UI/SwiftPlayground/main.swift

```diff
@@ -0,0 +1,32 @@
+//  Copyright © 2019 The nef Authors.
+
+import Foundation
+import CLIKit
+import nef
+
+let scriptName = "nef-swift-playground"
+let console = iPadConsole()
+
+func main() {
+    let args = arguments(keys: "package", "to", "name")
+    guard let packagePath = args["package"]?.expandingTildeInPath,
+          let outputPath = args["to"]?.expandingTildeInPath,
+          let projectName = args["name"] else {
+            Console.help.show(output: console)
+            exit(-1)
+    }
+    
+    guard let content = try? String(contentsOfFile: packagePath), !content.isEmpty else {
+        Console.error(information: "received an invalid Swift Package - '\(packagePath)'").show(output: console)
+        exit(-1)
+    }
+    
+    nef.SwiftPlayground.render(packageContent: content, name: projectName, output: URL(fileURLWithPath: outputPath))
+                       .provide(console)
+                       .unsafeRunSyncEither()
+                       .fold({ error in console.printError(information: ""); exit(-1) },
+                             { _ in console.printSuccess(); exit(0) })
```

Comment:
Ideally, these methods in `console` should return an `IO`, and they should run *before* the `unsafeRun` operation. `unsafeRun` should be the last thing in the program. Perhaps you can check the recently added `foldM` operation on `IO` and `EnvIO`.

Reply:
Yep, I agree, in this case, it is the "legacy" console pending to be refactored together the bash code, rest of the scripts... - `ConsoleOutput`. The new one is `NefModels.Console` and it is working exactly you said.

Anyway, I will update it as you commented to go in that direction - this is the first feature 100% FP